test_integration/test_scaling_region.py

```diff
@@ -0,0 +1,194 @@
+            freq_min = geopm_test_launcher.geopmread("CPUINFO::FREQ_MIN board 0")
+            freq_sticker = geopm_test_launcher.geopmread("CPUINFO::FREQ_STICKER board 0")
+            freq_step = geopm_test_launcher.geopmread("CPUINFO::FREQ_STEP board 0")
+            num_step = int((freq_sticker - freq_min) / freq_step + 0.5)
+            agent_conf_dict = {'FREQ_MIN':freq_min,           
+                               'FREQ_MAX':freq_sticker}
+            cls._region_freq = [freq_min + idx * freq_step
+                                for idx in range(num_step)]
+            freq_idx = 0
+            for freq_idx in range(len(cls._region_freq)):
+                agent_conf_dict['HASH_{}'.format(freq_idx)] = cls._region_hash[freq_idx]
+                agent_conf_dict['FREQ_{}'.format(freq_idx)] = cls._region_freq[freq_idx]
+            
+            agent_conf = geopmpy.io.AgentConf(cls._agent_conf_path,
+                                              'frequency_map',
+                                              agent_conf_dict)
+            launcher = geopm_test_launcher.TestLauncher(AppConf(),
+                                                        agent_conf,
+                                                        cls._report_path,
+                                                        cls._trace_path,
+                                                        time_limit=6000)
+            launcher.set_num_node(num_node)
+            launcher.set_num_rank(num_rank)
+            launcher.run(test_name)
+
+    @classmethod
+    def tearDownClass(cls):
+        """If we are not handling an exception and the GEOPM_KEEP_FILES
+        environment variable is unset, clean up output.
+
+        """
+        if (sys.exc_info() == (None, None, None) and not
+            cls._keep_files and not cls._skip_launch):
+            os.unlink(cls._agent_conf_path)
+            os.unlink(cls._report_path)
+
+    def test_monotone_performance(self):
+        """Test that the reports generated show linear performance with
+           respect to CPU frequency.
+
+        """
+        report = geopmpy.io.RawReport(self._report_path)
+        host_names = report.host_names()
+        for host in host_names:
+            region_names = [rn for rn in report.region_names(host)
+                            if rn.startswith('scaling_region')]
+            freq_time = []
+            for rn in region_names:
+                region = report.raw_region(host, rn)
+                freq = report.get_field(region, 'frequency', 'Hz')
+                time = report.get_field(region, 'runtime', 'sec')
+                freq_time.append((freq, time))
+            freq_time.sort()
+            time = list(zip(*freq_time)[1])
+            paired_time = zip(time[0:-1], time[1:])
+            for pt in paired_time:
+                self.assertGreater(pt[0], pt[1])
+
+if __name__ == '__main__':
+    try:
```

Comment:
remind me why we need this.  Doesn't setUpClass only get called once?

Reply:
The skip launch is to enable rerun of the tests without executing the benchmark.